examples/browser_on.js

```diff
@@ -0,0 +1,41 @@
+import launcher from 'k6/x/browser';
+import { check, sleep } from 'k6';
+
+export default function() {
+  const browser = launcher.launch('chromium', {
+    headless: __ENV.XK6_HEADLESS ? true : false,
+  });
+
+  check(browser, {
+    'should be connected after launch': browser.isConnected(),
+  });
+
+  sleep(2);
```

Comment:
Is this necessary here? Since this is an example, we'll guide the users to use `sleep` calls that we don't recommend.

Reply:
We discussed this previously, but we shouldn't suggest that `sleep()` is wrong for _all_ use cases. After all, if you genuinely want the script to wait for whatever reason (user pause, or in this case just waiting some time so that I could test `^C`), it should be fine to use. The "wrong" use case is the one where the script is waiting for an element to appear on the page, or any other browser event, for which we have the `waitFor*` methods, `waitUntil`, now `on(event)`, etc.

Reply:
I think people can see it and use it everywhere without thinking of that kind of stuff. But you know. I was wondering if it's essential there?

Reply:
People should read documentation, and I can add a comment here why it's used, but _we_ should be clear about what we recommend to users. Again, `sleep()` is perfectly valid in some use cases, and yes, it's essential here because I'm not quick enough to press `^C`. ;)

Reply:
I'm not arguing whether it's valid or not, for ^C, it's OK, I was asking for the merged example :-) No problemo. And this would be a good addition, +1:

> I can add a comment here why it's used

Reply:
I decided to remove `sleep()` from this script (7c332cb), since it was only useful for interactive testing the `^C` behavior, but since that's not useful for this yet, we should try to add integration tests for it by canceling the context, which will need to wait until https://github.com/grafana/k6/pull/2399 is merged.

Unfortunately, this exposed a nil panic issue which I tentatively fixed in a0e8d14. Let me know if you want a separate PR for that (and 02976ae and ce47855 I guess?). I'd rather keep them here if you don't mind; they're isolated commits, very small and easy to review.

And now I ran into #272... https://github.com/grafana/xk6-browser/runs/5498886146

Reply:
Is it better to check whether it's nil here by failing fast instead of silencing it? Besides, we now have a zombie FrameSession GR.

```go
func (fs *FrameSession) initFrameTree() error {
	...
	if frameTree, err = action2.Do(cdp.WithExecutor(fs.ctx, fs.session)); err != nil {
		return fmt.Errorf("unable to get page frame tree: %w", err)
	} else if frameTree == nil {
		...
	}

```

Reply:
I considered that, but returning an error here might have unintended side effects up the stack. Even though [the current test looks fine](https://github.com/grafana/xk6-browser/runs/5512995232), we know these are not exhaustive to be reliable.

And what zombie FrameSession GR? The only one launched is in `initEvents()`, which happens before this. Presumably they would all exit once `fs.ctx` is done, or just be reaped by Go on shutdown. We probably have dozens of those right now.

Anyway, since the test passes with that change, I left it in 51de7f9.